src/hotspot/share/opto/phaseX.hpp

```diff
@@ -528,7 +528,21 @@ class PhaseIterGVN : public PhaseGVN {
 
   // Add users of 'n' to worklist
   static void add_users_to_worklist0(Node* n, Unique_Node_List& worklist);
+
+  // Add one or more users of 'use' to the worklist if it appears that a
+  // known optimization could be applied to those users. Certain
+  // optimizations have dependencies that extend beyond a node's direct
+  // inputs, so it is necessary to ensure the appropriate notifications
+  // are made here.
```

Comment:
Maybe also add what 'n' is. I guess we could also rename it to 'parent'? But can also be done separately.